collector/main.go

```diff
@@ -97,10 +109,16 @@ func (lm *lifecycleManager) processEvents(ctx context.Context) {
 				return
 			}
 
+			err = lm.listener.Wait(ctx, res.RequestID)
```

Comment:
This will block indefinitely for a extensionAPI.Shutdown event, as that event does not carry a RequestID, so this call will wait for requestID zero to arrive, which never happens. (and the process will be killed by the environment) - 
need to move  the check for `res.EventType == extensionAPI.Shutdown` before the call to `lm.listener.Wait`